feat: Add team billing tables

joeauyeung opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- Adds `TeamBilling` and `OrganizationBilling` tables
- For new teams/orgs write the billing information to the corresponding tables

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

- Create a new team with billing enabled
    - The `TeamBilling` table should have the corresponding billing info
- Create a new org with billing enabled
    - The `OrganizationBilling` table should have the corresponding billing info

joeauyeung committed:
Init billing tables

joeauyeung committed:
Merge branch 'main' into add-team-billing-table

joeauyeung committed:
Create `IBillingRepository` types

joeauyeung committed:
Create billing repositories

joeauyeung committed:
Create billingRepositoryFactory

joeauyeung committed:
Eslint fix - remove unused organizationOnboarding

joeauyeung committed:
internal-team-billing create saveTeamBilling method using repositories

joeauyeung committed:
On new teams write to team billing table

joeauyeung committed:
On new org write to org billing table

joeauyeung committed:
Change fields to organizationId

joeauyeung committed:
Add todo comment

joeauyeung committed:
Revert "Change fields to organizationId"

This reverts commit bbb2e5dfa6b4c20a8a395f5730848a492cd70d68.

devin-ai-integration[bot] committed:
test: add comprehensive tests for team billing tables

- Fix credit-service.test.ts Prisma mock to export prisma object
- Replace any types with proper TypeScript types in credit-service.test.ts
- Add unit tests for PrismaTeamBillingRepository covering record creation, enum casting, and error handling
- Add unit tests for PrismaOrganizationBillingRepository with same coverage
- Add unit tests for BillingRepositoryFactory to verify correct repository selection
- Add unit tests for InternalTeamBilling.saveTeamBilling() method testing delegation to correct repositories
- All 53 tests pass with TZ=UTC yarn test
- Type checking passes with yarn type-check:ci --force

Co-Authored-By: joe@cal.com <j.auyeung419@gmail.com>

keithwillcode commented on packages/features/ee/billing/teams/internal-team-billing.test.ts:
```diff
@@ -173,4 +173,137 @@ describe("InternalTeamBilling", () => {
+        status: "ACTIVE" as const,
+      };
+
+      prismaMock.organizationBilling.create.mockResolvedValue({
```
Instead of introducing more usage of prismaMock, can we mock and verify the repository calls. Long-term plan is not to mock prisms directly. It’s a code smell.

devin-ai-integration[bot] committed:
refactor: update saveTeamBilling tests to mock repository interface

- Replace prismaMock usage with BillingRepositoryFactory mock
- Mock IBillingRepository interface instead of Prisma directly
- Follow repository mocking pattern from handleResponse.test.ts
- All tests passing (53 total)

Co-Authored-By: joe@cal.com <j.auyeung419@gmail.com>

joeauyeung committed:
Remove log statement

volnei commented on packages/features/ee/billing/repository/PrismaOrganizationBillingRepository.test.ts:
```diff
@@ -0,0 +1,178 @@
+import prismaMock from "../../../../../tests/libs/__mocks__/prismaMock";
```
anyway to avoid using prismaMock here?

volnei commented on packages/features/ee/billing/repository/PrismaTeamBillingRepository.test.ts:
```diff
@@ -0,0 +1,154 @@
+import prismaMock from "../../../../../tests/libs/__mocks__/prismaMock";
```
here too

volnei reviewed (changes requested):
We need to get rid of prismaMock.

joeauyeung committed:
Remove repository tests

joeauyeung commented:
@volnei removed the repository tests as these don't have any logic in them.

volnei reviewed (dismissed):
LGTM

joeauyeung committed:
Address feedback
